lib/api/core/models/repositories/profileRepository.js

```diff
@@ -130,7 +133,8 @@ module.exports = function (kuzzle) {
         // Fail if not all roles are found
         var rolesNotFound = _.difference(data.roles, extractRoleIds(roles));
         if (rolesNotFound.length) {
-          deferred.reject(new NotFoundError('Unable to hydrate profile with roles'));
+          deferred.reject(new NotFoundError('Unable to hydrate profile with roles: some roles don\'t exist.'));
```

Comment:
Nitpicking: to make the error message more explicit, we could add the non-existing roles. 
Something like this:
```js
new NotFoundError(`Unable to create profile ${requestObject.data._id}. The following roles don't exist: ${rolesNotFound}`)
```